packages/sanity/src/core/releases/components/documentHeader/dialog/CopyToNewReleaseDialog.tsx

```diff
@@ -43,25 +46,40 @@ export function CopyToNewReleaseDialog(props: {
     } as const
   })
   const [isSubmitting, setIsSubmitting] = useState(false)
+  const [_, setRerenderDialog] = useState(0)
 
   const telemetry = useTelemetry()
   const {createRelease} = useReleaseOperations()
 
   const displayTitle = title || t('release.placeholder-untitled-release')
 
+  const isScheduledDateInPast = getIsScheduledDateInPast(release)
+
   const handleOnChange = useCallback((changedValue: EditableReleaseDocument) => {
-    setValue(changedValue)
+    setRelease(changedValue)
   }, [])
 
   const handleAddVersion = useCallback(async () => {
     onCreateVersion(newReleaseId)
   }, [onCreateVersion, newReleaseId])
 
   const handleCreateRelease = useCallback(async () => {
+    // re-evaluate if date is in past
+    // as dialog could have been left idle for a while
+    if (getIsScheduledDateInPast(release)) {
+      toast.push({
+        closable: true,
+        status: 'warning',
+        title: tRelease('schedule-dialog.publish-date-in-past-warning'),
+      })
+      setRerenderDialog((cur) => cur + 1)
```

Comment:
ooh, do we need this? This feels like something of a hack 😬 
I don't understand why we need it, but I don't quite like it 🥲